pkgs/development/compilers/hip/hip-common.nix

```diff
@@ -0,0 +1,65 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, rocmUpdateScript
+, rocm-device-libs
+, rocm-runtime
+, rocminfo
+, lsb-release
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "hip-common";
+  version = "5.4.1";
+
+  src = fetchFromGitHub {
+    owner = "ROCm-Developer-Tools";
+    repo = "HIP";
+    rev = "rocm-${finalAttrs.version}";
+    hash = "sha256-JpHWTsR2Z8pXp1gNjO29pDYvH/cJvd5Dlpeig33UD28=";
+  };
+
+  postPatch = ''
+    substituteInPlace hip-lang-config.cmake.in \
+      --replace "\''${_IMPORT_PREFIX}/../llvm/lib/clang/*/include" "${stdenv.cc.cc}/lib/clang/*/include" \
+      --replace "\''${_IMPORT_PREFIX}/llvm/lib/clang/*/include" "${stdenv.cc.cc}/lib/clang/*/include" \
+      --replace "\''${ROCM_PATH}/include" "${rocm-runtime}/include" \
+      --replace "\''${HIP_COMPILER_INSTALL_PATH}/../lib/clang/*/lib/*" "${stdenv.cc.cc}/lib/clang/*/lib/*"
+
+    # The perl scripts will be replaced by HIPCC soon
+    substituteInPlace bin/hipcc.pl \
+      --replace "\$HIP_CLANG_PATH/../lib/clang/\$HIP_CLANG_VERSION" "${stdenv.cc}/resource-root" \
+      --replace "-L\$ROCM_PATH/lib" "${rocm-runtime}/lib" \
+      --replace "\''${ROCM_PATH}/bin/rocm_agent_enumerator" "${rocminfo}/bin/rocm_agent_enumerator"
+
+    substituteInPlace bin/hipconfig.pl \
+      --replace "/usr/bin/lsb_release" "${lsb-release}/bin/lsb_release"
+  '';
+
+  dontConfigure = true;
+  dontBuild = true;
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out
+    mv * $out
+
+    runHook postInstall
+  '';
+
+  passthru.updateScript = rocmUpdateScript {
+    name = finalAttrs.pname;
+    owner = finalAttrs.src.owner;
+    repo = finalAttrs.src.repo;
+  };
+
+  meta = with lib; {
+    description = "C++ Heterogeneous-Compute Interface for Portability";
+    homepage = "https://github.com/ROCm-Developer-Tools/HIP";
+    license = licenses.mit;
+    maintainers = with maintainers; [ lovesegfault ] ++ teams.rocm.members;
+    platforms = platforms.linux;
+    broken = finalAttrs.version != stdenv.cc.version;
```

Comment:
I am not sure if broken is actual correct here but I don't really have a better suggestion right now.

Reply:
The other option I found was including (llvmPackages_rocm) clang in the derivation, but since we already manually change the stdenv it feels like a needless point of redirection.